packages/flutter_tools/lib/src/artifacts.dart

```diff
@@ -299,3 +299,89 @@ class LocalEngineArtifacts extends Artifacts {
     throw Exception('Unsupported platform $platform.');
   }
 }
+
+/// A fuchsia host specific implementation of [Artifacts]
+///
+/// Certain binaries will have different locations which are not relative to
+/// a flutter tool directory.
+class FuchsiaHostArtifacts extends CachedArtifacts {
+  /// Create a [FuchsiaHostArtifacts] with the fuchsia host tools in `_fuchsiaHostTools`.
+  FuchsiaHostArtifacts(this._fuchsiaHostTools);
+
+  final Directory _fuchsiaHostTools;
+
+  @override
+  String getArtifactPath(Artifact artifact, [TargetPlatform platform, BuildMode mode]) {
+    platform ??= _currentHostPlatform;
+    switch (platform) {
+      case TargetPlatform.darwin_x64:
+      case TargetPlatform.linux_x64:
+      case TargetPlatform.fuchsia:
+        return _getHostArtifactPath(artifact, platform, mode);
+      case TargetPlatform.android_arm:
+      case TargetPlatform.android_arm64:
+      case TargetPlatform.android_x64:
+      case TargetPlatform.android_x86:
+      case TargetPlatform.windows_x64:
+      case TargetPlatform.tester:
+      case TargetPlatform.ios:
+        throw StateError('Invalid platform for fuchsia host');
+    }
+    assert(false, 'Invalid platform $platform.');
+    return null;
+  }
+
+  @override
+  String _getHostArtifactPath(Artifact artifact, TargetPlatform platform, BuildMode mode) {
+    switch (artifact) {
+      case Artifact.genSnapshot:
+        // For script snapshots any gen_snapshot binary will do. Returning gen_snapshot for
+        // android_arm in profile mode because it is available on all supported host platforms.
+        return _getAndroidArtifactPath(artifact, TargetPlatform.android_arm, BuildMode.profile);
```

Comment:
It's weird that we're calling `_getAndroidArtifactPath()` in a Fuchsia context.

Reply:
I don't think this is used but I was a bit anxious about removing it. I suppose It doesn't truly matter since this entrypoint is still unused.

Reply:
Removed for now.